apps/ui/Application.cpp

```diff
@@ -337,6 +337,8 @@ void Application::render()
 
     m_brayns.commitAndRender();
 
+    auto lock{m_brayns.getEngine().getRenderScopeLock()};
```

Comment:
This looks like a hack that will reduce maintainability. I leave it up to @tribal-tec to decide whether making the interface more function programming oriented would be better or not, but I would try to find another solution.

Reply:
Any comments here @tribal-tec?

Reply:
I don't see why the lock is needed where there is locking inside Brayns.cpp. But the whole change is suspicious to me, as suddenly an engine affects how things have to be done from the outside. I don't like in general how Brayns has to change to work with Optix. Ideally, all the hacks should be only within the engine implementation itself.